sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -815,6 +818,12 @@ case class AdaptiveSparkPlanExec(
     }
   }
 
+  private def assertStageNotFailed(stage: QueryStageExec): Unit = {
+    if (stage.hasFailed) {
+      throw stage.error.get().get
```

Comment:
Does this exception class and msg is same with the first one ? Shall we follow the behavior in `cleanUpAndThrowException` ?

How about holding a plan level exception? We now cache the plan for every query, it seems hold a plan level exception is reasonable (even without aqe).

Reply:
Updated to apply the same logic as `cleanUpAndThrowException`.

In terms of plan level exception, I think in AQE world it makes sense to cache the error since stages might be reused. In non-AQE world, a collect action will trigger a new job from a compiled RDD(the RDD is cached though). I'm not sure how do we want to define the error caching behavior...but it's definitely a good question worth further discussion. Can we address that in a follow up item?